conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -2063,29 +2078,53 @@ TEST_F(
           FAIL() << "zeEventQueryStatus() FAILED with " << eventResult;
         }
 
-        std::vector<uint8_t> rawData;
-        uint32_t rawDataSize = 0;
-        lzt::metric_streamer_read_data(metricStreamerHandle, rawDataSize,
-                                       &rawData);
-
-        rawData.resize(rawDataSize);
-        std::vector<zet_typed_value_t> metricValues;
-        std::vector<uint32_t> metricValueSets;
-        lzt::metric_calculate_metric_values_from_raw_data(
-            groupInfo.metricGroupHandle, rawData, metricValues,
-            metricValueSets);
-
-        std::vector<zet_metric_handle_t> metricHandles;
-        lzt::metric_get_metric_handles_from_metric_group(
-            groupInfo.metricGroupHandle, metricHandles);
-        std::vector<zet_metric_properties_t> metricProperties(
-            metricHandles.size());
-        lzt::metric_get_metric_properties_for_metric_group(metricHandles,
-                                                           metricProperties);
-
-        lzt::metric_validate_streamer_marker_data(metricProperties,
-                                                  metricValues, metricValueSets,
-                                                  streamerMarkerValues);
+        std::vector<uint8_t> rawData{};
+        uint32_t rawDataSize{};
+        const char *valueString =
+            std::getenv("LZT_METRIC_READ_DATA_MAX_DURATION_MS");
+        uint32_t maxDuration = 5; // millisecond(s)
+        if (valueString != nullptr) {
+          uint32_t value = atoi(valueString);
+          maxDuration = value != 0 ? value : maxDuration;
+          maxDuration = std::min(maxDuration, 100u);
+        }
+        auto start_time = std::chrono::high_resolution_clock::now();
+        auto duration = std::chrono::milliseconds(maxDuration);
+
+        while ((std::chrono::high_resolution_clock::now() - start_time) <
+               duration) {
+          std::vector<uint8_t> fetchedData{};
+          uint32_t fetchedDataSize{};
+          lzt::metric_streamer_read_data(metricStreamerHandle, fetchedDataSize,
```

Comment:
Cant we have a short sleep 1 ms sleep before we read the data every time ?

Reply:
Resolved, Thank you.